src/test/java/net/datafaker/service/WeightedRandomSelectorTest.java

```diff
@@ -0,0 +1,273 @@
+package net.datafaker.service;
+
+import net.datafaker.AbstractFakerTest;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.*;
+
+class WeightedRandomSelectorTest extends AbstractFakerTest {
+
+    public static final String WEIGHT_KEY = "weight";
+    public static final String VALUE_KEY = "value";
+    public static final String ELEMENT_1 = "Element1";
+    public static final String ELEMENT_2 = "Element2";
+    public static final String ELEMENT_3 = "Element3";
+    public static final String STRING_WEIGHT = "1.0";
+    public static final int ITERATIONS = 1000;
+    public static final double ELEMENT_1_WEIGHT = 1.0;
+    public static final double ELEMENT_2_WEIGHT = 2.0;
+    public static final double ELEMENT_3_WEIGHT = 3.0;
+    public static final double NEGATIVE_WEIGHT = -1.0;
+
+    /**
+     * Tests for exception scenarios using parameterized tests.
+     */
+    @ParameterizedTest
+    @MethodSource("exceptionCasesProvider")
+    void testWeightedArrayElement_exceptions(List<Map<String, Object>> items, String expectedMessage) {
```

Comment:
```suggestion
    void testWeightedArrayElementExceptions(List<Map<String, Object>> items, String expectedMessage) {
```
lets's do not mix different notations

Reply:
refactored